src/System.Private.CoreLib/shared/System/TimeZoneInfo.cs

```diff
@@ -1989,5 +1979,80 @@ private static void ValidateTimeZoneInfo(string id, TimeSpan baseUtcOffset, Adju
                 }
             }
         }
+
+        private static readonly TimeSpan MaxOffset = TimeSpan.FromHours(14.0);
+        private static readonly TimeSpan MinOffset = -MaxOffset;
+        
+        /// <summary>
+        /// Helper function that validates the TimeSpan is within +/- 14.0 hours
+        /// </summary>
+        internal static bool UtcOffsetOutOfRange(TimeSpan offset) =>
+            offset < MinOffset || offset > MaxOffset;
+
+        private static TimeSpan GetUtcOffset(TimeSpan baseUtcOffset, AdjustmentRule adjustmentRule)
+        {
+            return baseUtcOffset
+                + adjustmentRule.BaseUtcOffsetDelta
+                + (adjustmentRule.HasDaylightSaving ? adjustmentRule.DaylightDelta : TimeSpan.Zero);
```

Comment:
>adjustmentRule.HasDaylightSaving [](start = 19, length = 32)

do we really need this check? isn't DaylightDelta is Zero if we don't daylight saving?

Reply:
I would leave it in case it isn't - the cost is pretty much zero since we call it once per app